// @aws-sdk/client-comprehend/dist-cjs/Comprehend.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Comprehend = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const BatchDetectDominantLanguageCommand_1 = require("./commands/BatchDetectDominantLanguageCommand");
const BatchDetectEntitiesCommand_1 = require("./commands/BatchDetectEntitiesCommand");
const BatchDetectKeyPhrasesCommand_1 = require("./commands/BatchDetectKeyPhrasesCommand");
const BatchDetectSentimentCommand_1 = require("./commands/BatchDetectSentimentCommand");
const BatchDetectSyntaxCommand_1 = require("./commands/BatchDetectSyntaxCommand");
const BatchDetectTargetedSentimentCommand_1 = require("./commands/BatchDetectTargetedSentimentCommand");
const ClassifyDocumentCommand_1 = require("./commands/ClassifyDocumentCommand");
const ContainsPiiEntitiesCommand_1 = require("./commands/ContainsPiiEntitiesCommand");
const CreateDatasetCommand_1 = require("./commands/CreateDatasetCommand");
const CreateDocumentClassifierCommand_1 = require("./commands/CreateDocumentClassifierCommand");
const CreateEndpointCommand_1 = require("./commands/CreateEndpointCommand");
const CreateEntityRecognizerCommand_1 = require("./commands/CreateEntityRecognizerCommand");
const CreateFlywheelCommand_1 = require("./commands/CreateFlywheelCommand");
const DeleteDocumentClassifierCommand_1 = require("./commands/DeleteDocumentClassifierCommand");
const DeleteEndpointCommand_1 = require("./commands/DeleteEndpointCommand");
const DeleteEntityRecognizerCommand_1 = require("./commands/DeleteEntityRecognizerCommand");
const DeleteFlywheelCommand_1 = require("./commands/DeleteFlywheelCommand");
const DeleteResourcePolicyCommand_1 = require("./commands/DeleteResourcePolicyCommand");
const DescribeDatasetCommand_1 = require("./commands/DescribeDatasetCommand");
const DescribeDocumentClassificationJobCommand_1 = require("./commands/DescribeDocumentClassificationJobCommand");
const DescribeDocumentClassifierCommand_1 = require("./commands/DescribeDocumentClassifierCommand");
const DescribeDominantLanguageDetectionJobCommand_1 = require("./commands/DescribeDominantLanguageDetectionJobCommand");
const DescribeEndpointCommand_1 = require("./commands/DescribeEndpointCommand");
const DescribeEntitiesDetectionJobCommand_1 = require("./commands/DescribeEntitiesDetectionJobCommand");
const DescribeEntityRecognizerCommand_1 = require("./commands/DescribeEntityRecognizerCommand");
const DescribeEventsDetectionJobCommand_1 = require("./commands/DescribeEventsDetectionJobCommand");
const DescribeFlywheelCommand_1 = require("./commands/DescribeFlywheelCommand");
const DescribeFlywheelIterationCommand_1 = require("./commands/DescribeFlywheelIterationCommand");
const DescribeKeyPhrasesDetectionJobCommand_1 = require("./commands/DescribeKeyPhrasesDetectionJobCommand");
const DescribePiiEntitiesDetectionJobCommand_1 = require("./commands/DescribePiiEntitiesDetectionJobCommand");
const DescribeResourcePolicyCommand_1 = require("./commands/DescribeResourcePolicyCommand");
const DescribeSentimentDetectionJobCommand_1 = require("./commands/DescribeSentimentDetectionJobCommand");
const DescribeTargetedSentimentDetectionJobCommand_1 = require("./commands/DescribeTargetedSentimentDetectionJobCommand");
const DescribeTopicsDetectionJobCommand_1 = require("./commands/DescribeTopicsDetectionJobCommand");
const DetectDominantLanguageCommand_1 = require("./commands/DetectDominantLanguageCommand");
const DetectEntitiesCommand_1 = require("./commands/DetectEntitiesCommand");
const DetectKeyPhrasesCommand_1 = require("./commands/DetectKeyPhrasesCommand");
const DetectPiiEntitiesCommand_1 = require("./commands/DetectPiiEntitiesCommand");
const DetectSentimentCommand_1 = require("./commands/DetectSentimentCommand");
const DetectSyntaxCommand_1 = require("./commands/DetectSyntaxCommand");
const DetectTargetedSentimentCommand_1 = require("./commands/DetectTargetedSentimentCommand");
const ImportModelCommand_1 = require("./commands/ImportModelCommand");
const ListDatasetsCommand_1 = require("./commands/ListDatasetsCommand");
const ListDocumentClassificationJobsCommand_1 = require("./commands/ListDocumentClassificationJobsCommand");
const ListDocumentClassifiersCommand_1 = require("./commands/ListDocumentClassifiersCommand");
const ListDocumentClassifierSummariesCommand_1 = require("./commands/ListDocumentClassifierSummariesCommand");
const ListDominantLanguageDetectionJobsCommand_1 = require("./commands/ListDominantLanguageDetectionJobsCommand");
const ListEndpointsCommand_1 = require("./commands/ListEndpointsCommand");
const ListEntitiesDetectionJobsCommand_1 = require("./commands/ListEntitiesDetectionJobsCommand");
const ListEntityRecognizersCommand_1 = require("./commands/ListEntityRecognizersCommand");
const ListEntityRecognizerSummariesCommand_1 = require("./commands/ListEntityRecognizerSummariesCommand");
const ListEventsDetectionJobsCommand_1 = require("./commands/ListEventsDetectionJobsCommand");
const ListFlywheelIterationHistoryCommand_1 = require("./commands/ListFlywheelIterationHistoryCommand");
const ListFlywheelsCommand_1 = require("./commands/ListFlywheelsCommand");
const ListKeyPhrasesDetectionJobsCommand_1 = require("./commands/ListKeyPhrasesDetectionJobsCommand");
const ListPiiEntitiesDetectionJobsCommand_1 = require("./commands/ListPiiEntitiesDetectionJobsCommand");
const ListSentimentDetectionJobsCommand_1 = require("./commands/ListSentimentDetectionJobsCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ListTargetedSentimentDetectionJobsCommand_1 = require("./commands/ListTargetedSentimentDetectionJobsCommand");
const ListTopicsDetectionJobsCommand_1 = require("./commands/ListTopicsDetectionJobsCommand");
const PutResourcePolicyCommand_1 = require("./commands/PutResourcePolicyCommand");
const StartDocumentClassificationJobCommand_1 = require("./commands/StartDocumentClassificationJobCommand");
const StartDominantLanguageDetectionJobCommand_1 = require("./commands/StartDominantLanguageDetectionJobCommand");
const StartEntitiesDetectionJobCommand_1 = require("./commands/StartEntitiesDetectionJobCommand");
const StartEventsDetectionJobCommand_1 = require("./commands/StartEventsDetectionJobCommand");
const StartFlywheelIterationCommand_1 = require("./commands/StartFlywheelIterationCommand");
const StartKeyPhrasesDetectionJobCommand_1 = require("./commands/StartKeyPhrasesDetectionJobCommand");
const StartPiiEntitiesDetectionJobCommand_1 = require("./commands/StartPiiEntitiesDetectionJobCommand");
const StartSentimentDetectionJobCommand_1 = require("./commands/StartSentimentDetectionJobCommand");
const StartTargetedSentimentDetectionJobCommand_1 = require("./commands/StartTargetedSentimentDetectionJobCommand");
const StartTopicsDetectionJobCommand_1 = require("./commands/StartTopicsDetectionJobCommand");
const StopDominantLanguageDetectionJobCommand_1 = require("./commands/StopDominantLanguageDetectionJobCommand");
const StopEntitiesDetectionJobCommand_1 = require("./commands/StopEntitiesDetectionJobCommand");
const StopEventsDetectionJobCommand_1 = require("./commands/StopEventsDetectionJobCommand");
const StopKeyPhrasesDetectionJobCommand_1 = require("./commands/StopKeyPhrasesDetectionJobCommand");
const StopPiiEntitiesDetectionJobCommand_1 = require("./commands/StopPiiEntitiesDetectionJobCommand");
const StopSentimentDetectionJobCommand_1 = require("./commands/StopSentimentDetectionJobCommand");
const StopTargetedSentimentDetectionJobCommand_1 = require("./commands/StopTargetedSentimentDetectionJobCommand");
const StopTrainingDocumentClassifierCommand_1 = require("./commands/StopTrainingDocumentClassifierCommand");
const StopTrainingEntityRecognizerCommand_1 = require("./commands/StopTrainingEntityRecognizerCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateEndpointCommand_1 = require("./commands/UpdateEndpointCommand");
const UpdateFlywheelCommand_1 = require("./commands/UpdateFlywheelCommand");
const ComprehendClient_1 = require("./ComprehendClient");
const commands = {
    BatchDetectDominantLanguageCommand: BatchDetectDominantLanguageCommand_1.BatchDetectDominantLanguageCommand,
    BatchDetectEntitiesCommand: BatchDetectEntitiesCommand_1.BatchDetectEntitiesCommand,
    BatchDetectKeyPhrasesCommand: BatchDetectKeyPhrasesCommand_1.BatchDetectKeyPhrasesCommand,
    BatchDetectSentimentCommand: BatchDetectSentimentCommand_1.BatchDetectSentimentCommand,
    BatchDetectSyntaxCommand: BatchDetectSyntaxCommand_1.BatchDetectSyntaxCommand,
    BatchDetectTargetedSentimentCommand: BatchDetectTargetedSentimentCommand_1.BatchDetectTargetedSentimentCommand,
    ClassifyDocumentCommand: ClassifyDocumentCommand_1.ClassifyDocumentCommand,
    ContainsPiiEntitiesCommand: ContainsPiiEntitiesCommand_1.ContainsPiiEntitiesCommand,
    CreateDatasetCommand: CreateDatasetCommand_1.CreateDatasetCommand,
    CreateDocumentClassifierCommand: CreateDocumentClassifierCommand_1.CreateDocumentClassifierCommand,
    CreateEndpointCommand: CreateEndpointCommand_1.CreateEndpointCommand,
    CreateEntityRecognizerCommand: CreateEntityRecognizerCommand_1.CreateEntityRecognizerCommand,
    CreateFlywheelCommand: CreateFlywheelCommand_1.CreateFlywheelCommand,
    DeleteDocumentClassifierCommand: DeleteDocumentClassifierCommand_1.DeleteDocumentClassifierCommand,
    DeleteEndpointCommand: DeleteEndpointCommand_1.DeleteEndpointCommand,
    DeleteEntityRecognizerCommand: DeleteEntityRecognizerCommand_1.DeleteEntityRecognizerCommand,
    DeleteFlywheelCommand: DeleteFlywheelCommand_1.DeleteFlywheelCommand,
    DeleteResourcePolicyCommand: DeleteResourcePolicyCommand_1.DeleteResourcePolicyCommand,
    DescribeDatasetCommand: DescribeDatasetCommand_1.DescribeDatasetCommand,
    DescribeDocumentClassificationJobCommand: DescribeDocumentClassificationJobCommand_1.DescribeDocumentClassificationJobCommand,
    DescribeDocumentClassifierCommand: DescribeDocumentClassifierCommand_1.DescribeDocumentClassifierCommand,
    DescribeDominantLanguageDetectionJobCommand: DescribeDominantLanguageDetectionJobCommand_1.DescribeDominantLanguageDetectionJobCommand,
    DescribeEndpointCommand: DescribeEndpointCommand_1.DescribeEndpointCommand,
    DescribeEntitiesDetectionJobCommand: DescribeEntitiesDetectionJobCommand_1.DescribeEntitiesDetectionJobCommand,
    DescribeEntityRecognizerCommand: DescribeEntityRecognizerCommand_1.DescribeEntityRecognizerCommand,
    DescribeEventsDetectionJobCommand: DescribeEventsDetectionJobCommand_1.DescribeEventsDetectionJobCommand,
    DescribeFlywheelCommand: DescribeFlywheelCommand_1.DescribeFlywheelCommand,
    DescribeFlywheelIterationCommand: DescribeFlywheelIterationCommand_1.DescribeFlywheelIterationCommand,
    DescribeKeyPhrasesDetectionJobCommand: DescribeKeyPhrasesDetectionJobCommand_1.DescribeKeyPhrasesDetectionJobCommand,
    DescribePiiEntitiesDetectionJobCommand: DescribePiiEntitiesDetectionJobCommand_1.DescribePiiEntitiesDetectionJobCommand,
    DescribeResourcePolicyCommand: DescribeResourcePolicyCommand_1.DescribeResourcePolicyCommand,
    DescribeSentimentDetectionJobCommand: DescribeSentimentDetectionJobCommand_1.DescribeSentimentDetectionJobCommand,
    DescribeTargetedSentimentDetectionJobCommand: DescribeTargetedSentimentDetectionJobCommand_1.DescribeTargetedSentimentDetectionJobCommand,
    DescribeTopicsDetectionJobCommand: DescribeTopicsDetectionJobCommand_1.DescribeTopicsDetectionJobCommand,
    DetectDominantLanguageCommand: DetectDominantLanguageCommand_1.DetectDominantLanguageCommand,
    DetectEntitiesCommand: DetectEntitiesCommand_1.DetectEntitiesCommand,
    DetectKeyPhrasesCommand: DetectKeyPhrasesCommand_1.DetectKeyPhrasesCommand,
    DetectPiiEntitiesCommand: DetectPiiEntitiesCommand_1.DetectPiiEntitiesCommand,
    DetectSentimentCommand: DetectSentimentCommand_1.DetectSentimentCommand,
    DetectSyntaxCommand: DetectSyntaxCommand_1.DetectSyntaxCommand,
    DetectTargetedSentimentCommand: DetectTargetedSentimentCommand_1.DetectTargetedSentimentCommand,
    ImportModelCommand: ImportModelCommand_1.ImportModelCommand,
    ListDatasetsCommand: ListDatasetsCommand_1.ListDatasetsCommand,
    ListDocumentClassificationJobsCommand: ListDocumentClassificationJobsCommand_1.ListDocumentClassificationJobsCommand,
    ListDocumentClassifiersCommand: ListDocumentClassifiersCommand_1.ListDocumentClassifiersCommand,
    ListDocumentClassifierSummariesCommand: ListDocumentClassifierSummariesCommand_1.ListDocumentClassifierSummariesCommand,
    ListDominantLanguageDetectionJobsCommand: ListDominantLanguageDetectionJobsCommand_1.ListDominantLanguageDetectionJobsCommand,
    ListEndpointsCommand: ListEndpointsCommand_1.ListEndpointsCommand,
    ListEntitiesDetectionJobsCommand: ListEntitiesDetectionJobsCommand_1.ListEntitiesDetectionJobsCommand,
    ListEntityRecognizersCommand: ListEntityRecognizersCommand_1.ListEntityRecognizersCommand,
    ListEntityRecognizerSummariesCommand: ListEntityRecognizerSummariesCommand_1.ListEntityRecognizerSummariesCommand,
    ListEventsDetectionJobsCommand: ListEventsDetectionJobsCommand_1.ListEventsDetectionJobsCommand,
    ListFlywheelIterationHistoryCommand: ListFlywheelIterationHistoryCommand_1.ListFlywheelIterationHistoryCommand,
    ListFlywheelsCommand: ListFlywheelsCommand_1.ListFlywheelsCommand,
    ListKeyPhrasesDetectionJobsCommand: ListKeyPhrasesDetectionJobsCommand_1.ListKeyPhrasesDetectionJobsCommand,
    ListPiiEntitiesDetectionJobsCommand: ListPiiEntitiesDetectionJobsCommand_1.ListPiiEntitiesDetectionJobsCommand,
    ListSentimentDetectionJobsCommand: ListSentimentDetectionJobsCommand_1.ListSentimentDetectionJobsCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ListTargetedSentimentDetectionJobsCommand: ListTargetedSentimentDetectionJobsCommand_1.ListTargetedSentimentDetectionJobsCommand,
    ListTopicsDetectionJobsCommand: ListTopicsDetectionJobsCommand_1.ListTopicsDetectionJobsCommand,
    PutResourcePolicyCommand: PutResourcePolicyCommand_1.PutResourcePolicyCommand,
    StartDocumentClassificationJobCommand: StartDocumentClassificationJobCommand_1.StartDocumentClassificationJobCommand,
    StartDominantLanguageDetectionJobCommand: StartDominantLanguageDetectionJobCommand_1.StartDominantLanguageDetectionJobCommand,
    StartEntitiesDetectionJobCommand: StartEntitiesDetectionJobCommand_1.StartEntitiesDetectionJobCommand,
    StartEventsDetectionJobCommand: StartEventsDetectionJobCommand_1.StartEventsDetectionJobCommand,
    StartFlywheelIterationCommand: StartFlywheelIterationCommand_1.StartFlywheelIterationCommand,
    StartKeyPhrasesDetectionJobCommand: StartKeyPhrasesDetectionJobCommand_1.StartKeyPhrasesDetectionJobCommand,
    StartPiiEntitiesDetectionJobCommand: StartPiiEntitiesDetectionJobCommand_1.StartPiiEntitiesDetectionJobCommand,
    StartSentimentDetectionJobCommand: StartSentimentDetectionJobCommand_1.StartSentimentDetectionJobCommand,
    StartTargetedSentimentDetectionJobCommand: StartTargetedSentimentDetectionJobCommand_1.StartTargetedSentimentDetectionJobCommand,
    StartTopicsDetectionJobCommand: StartTopicsDetectionJobCommand_1.StartTopicsDetectionJobCommand,
    StopDominantLanguageDetectionJobCommand: StopDominantLanguageDetectionJobCommand_1.StopDominantLanguageDetectionJobCommand,
    StopEntitiesDetectionJobCommand: StopEntitiesDetectionJobCommand_1.StopEntitiesDetectionJobCommand,
    StopEventsDetectionJobCommand: StopEventsDetectionJobCommand_1.StopEventsDetectionJobCommand,
    StopKeyPhrasesDetectionJobCommand: StopKeyPhrasesDetectionJobCommand_1.StopKeyPhrasesDetectionJobCommand,
    StopPiiEntitiesDetectionJobCommand: StopPiiEntitiesDetectionJobCommand_1.StopPiiEntitiesDetectionJobCommand,
    StopSentimentDetectionJobCommand: StopSentimentDetectionJobCommand_1.StopSentimentDetectionJobCommand,
    StopTargetedSentimentDetectionJobCommand: StopTargetedSentimentDetectionJobCommand_1.StopTargetedSentimentDetectionJobCommand,
    StopTrainingDocumentClassifierCommand: StopTrainingDocumentClassifierCommand_1.StopTrainingDocumentClassifierCommand,
    StopTrainingEntityRecognizerCommand: StopTrainingEntityRecognizerCommand_1.StopTrainingEntityRecognizerCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateEndpointCommand: UpdateEndpointCommand_1.UpdateEndpointCommand,
    UpdateFlywheelCommand: UpdateFlywheelCommand_1.UpdateFlywheelCommand,
};
class Comprehend extends ComprehendClient_1.ComprehendClient {
}
exports.Comprehend = Comprehend;
(0, smithy_client_1.createAggregatedClient)(commands, Comprehend);
