// @aws-sdk/client-transcribe-streaming/dist-es/commands/StartCallAnalyticsStreamTranscriptionCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getEventStreamPlugin } from "@aws-sdk/middleware-eventstream";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { getWebSocketPlugin } from "@aws-sdk/middleware-websocket";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartCallAnalyticsStreamTranscriptionRequestFilterSensitiveLog, StartCallAnalyticsStreamTranscriptionResponseFilterSensitiveLog, } from "../models/models_0";
import { de_StartCallAnalyticsStreamTranscriptionCommand, se_StartCallAnalyticsStreamTranscriptionCommand, } from "../protocols/Aws_restJson1";
export class StartCallAnalyticsStreamTranscriptionCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, StartCallAnalyticsStreamTranscriptionCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getEventStreamPlugin(configuration));
        this.middlewareStack.use(getWebSocketPlugin(configuration, { headerPrefix: "x-amzn-transcribe-" }));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "TranscribeStreamingClient";
        const commandName = "StartCallAnalyticsStreamTranscriptionCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StartCallAnalyticsStreamTranscriptionRequestFilterSensitiveLog,
            outputFilterSensitiveLog: StartCallAnalyticsStreamTranscriptionResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_StartCallAnalyticsStreamTranscriptionCommand(input, context);
    }
    deserialize(output, context) {
        return de_StartCallAnalyticsStreamTranscriptionCommand(output, context);
    }
}
