// @aws-sdk/client-comprehend/dist-es/commands/DetectDominantLanguageCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DetectDominantLanguageRequestFilterSensitiveLog, DetectDominantLanguageResponseFilterSensitiveLog, } from "../models/models_0";
import { de_DetectDominantLanguageCommand, se_DetectDominantLanguageCommand } from "../protocols/Aws_json1_1";
export class DetectDominantLanguageCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, DetectDominantLanguageCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "ComprehendClient";
        const commandName = "DetectDominantLanguageCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: DetectDominantLanguageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DetectDominantLanguageResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_DetectDominantLanguageCommand(input, context);
    }
    deserialize(output, context) {
        return de_DetectDominantLanguageCommand(output, context);
    }
}
